buildconfig/CMake/Eigen.cmake

```diff
@@ -8,7 +8,8 @@ option(USE_SYSTEM_EIGEN "Use the system installed Eigen - v${eigen_version}?" OF
 
 if(USE_SYSTEM_EIGEN)
   message(STATUS "Using system Eigen")
-  find_package(Eigen ${eigen_version} EXACT REQUIRED)
+  find_package(Eigen3 ${eigen_version} EXACT REQUIRED
```

Comment:
I think the problem on Ubuntu is that we'll never be able to match the version exactly. I have `3.2.92` on `16.04`.

Also it doesn't actually seem to pick up the version of the system one correctly anyway:

```
The following configuration files were considered but not accepted:

    /usr/lib/cmake/eigen3/Eigen3Config.cmake, version: unknown
```

I think that's a problem with the `Eigen3` cmake config that doesn't set a `EIGEN3_VERSION` variable that cmake expects.

Shall we just drop the version requirement?